src/app/services/goals-data-service.service.ts

```diff
@@ -42,6 +94,133 @@ export class GoalsDataService {
     );
   }
 
+  getMostRecentObservationResult(patientId: string, code: string): Observable<MccObservation> {
+    const url = `${environment.mccapiUrl}${this.observationURL}?subject=${patientId}&code=${code}`;
+    return this.http.get<MccObservation>(url, this.httpOptions).pipe(
+      tap(_ => this.log(`fetched MccObservation patientId=${patientId} code=${code}`)),
+      catchError(this.handleError<MccObservation>(`getMostRecentObservationResult patientId=${patientId} code=${code}`))
+    );
+  }
+
+
+  formatTargetValue(target: GoalTarget, mostRecentResultValue: string): any[] {
```

Comment:
Would this function not be better as a general utility

Reply:
Yes.  I created src/utility-functions.ts  and moved the function into that file.  I have pushed this change and it is in this pull request.